### DISCUSSION TITLE
Support for hashrefs hidding inside of arrays

### DISCUSSION THREAD

**Discussion by oodler577** [haukex/Util-H2O#16] [Ideas]
@haukex I created this issue in the `Util::H2O::More` repo, but I wanted to get some input - I really think it should be a modifier similar to `-recurse` but something that implies it'll do a full tree search of the ref (`HASH` or `ARRAY`) and turn all `HASH` refs it finds into a `bless`'d reference with accessors. I am finding a lot of `JSON` APIs present their results as _arrays of hashes_ (when results are record oriented, which is often). 

Would something like this fit in `Util::H2O` more than `::More`?

https://github.com/oodler577/p5-Util-H2O-More/issues/5

**Comment by oodler577** [comment]
Oh, with complementary support with something like `o2h` to undo it.

**Comment by oodler577** [comment]
Another use case would be to allow for  h2o`'ing contents of `@_` in a `sub`, but that might require `h2o` to ignore some stuff;

consider,

```
sub foo {
  my ($self, $bar, $baz) = h2o -list-mode, @_;
  ..
  return $bar->some_attribute * $baz->some_other_attribute;
}
```

But `$self` would need to be ignored; main point is to use `h2o` as a filter for parameter lists, acting upon them only if it would have normally. I had not thought of this general application, but `h2o -some-modifier ARRAY` seems to be an interesting and useful idiom.

**Comment by haukex** [comment]
See #18 !
